invenio_users_resources/services/users/config.py

```diff
@@ -68,9 +68,14 @@ class UserSearchOptions(SearchOptions, SearchOptionsMixin):
     # The user search needs to be highly restricted to avoid leaking
     # account information, hence do not edit here unless you are
     # absolutely sure what you're doing.
-    suggest_parser_cls = SuggestQueryParser.factory(
+    suggest_parser_cls = CompositeSuggestQueryParser.factory(
         tree_transformer_cls=SearchFieldTransformer,
-        fields=["username^2", "email^2", "profile.full_name^3", "profile.affiliations"],
+        fields=[
+            "username^2",
```

Comment:
:warning:
Searching for a username with a dash is still not working well.
For instance, searching from "one-two" seems to search for usernames starting with "one" and starting with "two", and therefore does not find anything.

Reply:
I think this is the issue, text field is split by `-` https://github.com/inveniosoftware/invenio-users-resources/blob/master/invenio_users_resources/records/mappings/os-v2/users/user-v3.0.0.json#L132
if you search by username.keyword, it should work